lib/diaspora/shareable.rb

```diff
@@ -57,7 +65,7 @@ def owned_or_visible_by_user(user)
         with_visibility.where(
           visible_by_user(user).or(arel_table[:public].eq(true)
                                      .or(arel_table[:author_id].eq(user.person_id)))
-        ).select("DISTINCT #{table_name}.*")
+        ).distinct_wrapper
```

Comment:
`.count` called on the relation doesn't work if `.select("DISTINCT #{table_name}.*")` used with mysql. `.distinct` fixes the thing for mysql, but breaks for postgres. You can reproduce this failure by reverting the change and running tests for `spec/models/status_message_spec.rb`.